cro-clib/chain.h

```diff
@@ -0,0 +1,65 @@
+#include <stdarg.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+#define ENCKEY_LENGTH 32
+
+#define MNEMONIC_LENGTH 512
+
+#define SEED_LENGTH 64
+
+typedef struct HDWallet HDWallet;
+
+typedef HDWallet *HDWalletPtr;
+
+typedef struct HDWalletCreateResult {
+  int32_t error;
+  uint8_t mnemonic[MNEMONIC_LENGTH];
+  uint32_t mnemonic_length;
+} HDWalletCreateResult;
+
+/**
+ * # Safety
+ * this function should not be called inside rust
+ */
+int32_t create_hdwallet(const char *_name,
+                        const char *_passphrase,
+                        HDWalletPtr *wallet_out,
+                        HDWalletCreateResult *create_result);
```

Comment:
Why not `char* mnemonic_buf, size_t len`.

Reply:
ok,i will change it

Reply:
For `\0`-terminated strings -- use this type: https://doc.rust-lang.org/std/ffi/struct.CString.html

Reply:
ok